aggregator/src/main/scala/weco/concepts/aggregator/secrets/ClusterConfWithSecrets.scala

```diff
@@ -0,0 +1,39 @@
+package weco.concepts.aggregator.secrets
+
+import grizzled.slf4j.Logging
+import weco.concepts.common.elasticsearch.Indexer
+
+/*
+ * A Resolver for ClusterConfig secrets.
+ * If a ClusterConfig contains _references to_ secret values,
+ * this class replaces those references with the actual secrets.
+ *
+ * The caller must provide a resolver that can look up the secrets and
+ * return them in a map.
+ *
+ */
+class ClusterConfWithSecrets(
+  resolver: Seq[String] => Map[String, String]
+) extends Logging {
+  def apply(
+    clusterConfig: Indexer.ClusterConfig
+  ): Indexer.ClusterConfig = {
+    clusterConfig match {
+      case Indexer.ClusterConfig(host, _, _, _, Some(password), true) =>
```

Comment:
Basically, I know that these two are the two "secret" properties.  I could make this a bit more general purpose by somehow recognising whether a value is a secret key or the actual value to use, but I think this is sufficient for this application.